README.md

```diff
@@ -2,68 +2,76 @@
 
 # GraphQL schema components.
 
-This project is designed to make npm module or component based Node.js development of graphql schemas easy.
+This project is designed to faciliate componentized or modularized development of GraphQL schemas.
 
 Read more about the idea [here](https://medium.com/expedia-group-tech/graphql-component-architecture-principles-homeaway-ede8a58d6fde).
 
-`graphql-component` lets you built a schema progressively through a tree of graphql schema dependencies.
+`graphql-component` lets you build a schema progressively through a tree (faciliated through `imports`) of GraphQLComponent instances. Each GraphQLComponent instance encapsulates an executable GraphQL schema, specifically a `graphql-js` GraphQLSchema object. See the API below, but the encapsulated schema is accessible through a simple `schema` getter on a given `GraphQLComponent` instance.
 
-### Repository structure
+Generally speaking, each instance of `GraphQLComponent` has reference to an instance of [`GraphQLSchema`](https://graphql.org/graphql-js/type/#graphqlschema). This instance of `GraphQLSchema` is built in a several ways, depending on the options passed to a given `GraphQLComponent`'s constructor.
 
-- `lib` - the graphql-component code.
-- `test/examples/example-listing/property-component` - a component implementation for `Property`.
-- `test/examples/example-listing/reviews-component` - a component implementation for `Reviews`.
-- `test/examples/example-listing/listing-component` - a component implementation composing `Property` and `Reviews` into a new `Listing`.
-- `test/examples/example-listing/server` - the "application".
+* when a `GraphQLComponent` instance has `imports` (ie. other `GraphQLComponent` instances or component configuration objects) [graphql-tools stitchSchemas()](https://www.graphql-tools.com/docs/schema-stitching/) is used to create a "gateway" or aggregate schema that is the combination of the underlying imported schemas, and the typeDefs/resolvers passed to the root or importing `GraphQLComponent`
+* when a `GraphQLComponent` has no imports, graphql-tools' `makeExecuteableSchema({typeDefs, resolvers})` is used to generate an executable GraphQL schema using the passed/required inputs.
+
+It's worth noting that `GraphQLComponent` can also be used to construct componentized Apollo Federated schemas. That is, if you pass the `federation: true` flag to a GraphQLComponent constructor, `@apollo/federation`'s [buildFederatedSchema()]() is used in lieu of graphql-tools `makeExecutableSchema({...})` and the above still schema construction rule applies. The general use case here might be to help modularize an individual federated subschema service implementation.
```

Comment:
I guess I never got around to linking the buildFederatedSchema() function to code or docs - probably just best to remove the markdown to make this a link since one can simply follow our package.json to this function if needed.